arrow/repeat.go

```diff
@@ -10,60 +12,202 @@ import (
 	"github.com/influxdata/flux/values"
 )
 
-// Repeat will construct an arrow array that repeats
-// the value n times.
+// Repeat will construct an arrow array that repeats the value n times.
 func Repeat(v values.Value, n int, mem memory.Allocator) array.Interface {
```

Comment:
I wonder if this could be taken a step further and *never* materialize an array for columns that have all the same value?  Since you could just store a scalar value that would be the value returned for any call to `Value`?